tests/zfs-tests/include/libtest.shlib

```diff
@@ -3366,6 +3366,75 @@ function tunable_exists
 	get_tunable_impl $1 "zfs" 1
 }
 
+function save_tunable64
+{
+	save_tunable_impl "$1" Z
+}
+
+function save_tunable32
+{
+	save_tunable_impl "$1" W
+}
+
+typeset -A libtest_saved_tunables
+function save_tunable_impl
+{
+	typeset tunable="$1"
+	typeset mdb_cmd="$2"
+	typeset module="${3:-zfs}"
+
+	# We key with $mdb_cmd to detect an incorrect pair of
+	# save_tunable and restore_tunable.
+	# For example, `save_tunable64 X` followed by `restore_tunable32 X`
+	# should fail with error "Tunable has not been saved"
+	if [ -n "${libtest_saved_tunables[$module/$mdb_cmd/$tunable]}" ]; then
+		log_fail "Tunable already saved: $tunable [module=$module, mdb_cmd=$mdb_cmd]"
+	fi
+
+	typeset current_value
+	current_value="$(get_tunable_impl "$tunable" "$module")"
+	if [ "$?" != 0 ]; then
```

Comment:
`c_v="$()" || l_f ...`, likewise below

Reply:
I don't know if that actually improves readability, considering how long the `log_fail` message is.
1. Should I still do it?
2. What's the convention for turning a `v="$(...)" || log_fail ...` statement into multiple lines?